pkg/service/issuing/model.go

```diff
@@ -0,0 +1,149 @@
+package issuing
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/tbd54566975/ssi-service/config"
+	"github.com/tbd54566975/ssi-service/pkg/service/framework"
+	"github.com/tbd54566975/ssi-service/pkg/storage"
+	"go.einride.tech/aip/filtering"
+)
+
+type GetIssuanceTemplateRequest struct {
+	ID string `json:"id" validate:"required"`
+}
+
+type CredentialTemplateData struct {
+	// ID of the input descriptor in the application. Correponds to one of the
+	// PresentationDefinition.InputDescriptors[].ID in the credential manifest.
+	CredentialInputDescriptor string `json:"credentialInputDescriptor"`
+
+	// The set of information that will be used to create claims.
+	Claims ClaimTemplates
+}
+
+type TimeLike struct {
+	// For fixed time in the future.
+	*time.Time
+
+	// For a fixed offset from when it was issued.
+	*time.Duration
+}
+
+type ClaimTemplates struct {
+	// Values may be json path like strings, or any other JSON primitive. Each entry will be used to come up with a
+	// claim about the credentialSubject in the credential that will be issued.
+	Data map[string]any
+}
+
+type CredentialTemplate struct {
+	// ID corresponding to an OutputDescriptor.ID from the manifest.
+	ID string `json:"id"`
+
+	// ID of the CredentialSchema to be used for the issued credential.
+	Schema string `json:"schema"`
+
+	// Date that will be used to determine credential claims.
+	Data CredentialTemplateData `json:"data"`
+
+	// Parameter to determine the expiry of the credential.
+	Expiry TimeLike `json:"expiry"`
+
+	// Whether the credentials created should be revocable.
+	Revocable bool `json:"revocable"`
+}
+
+type IssuanceTemplate struct {
+	// ID of the credential manifest that this template corresponds to.
+	CredentialManifest string `json:"credentialManifest"`
+
+	// ID of the issuer that will be issuing the credentials.
+	Issuer string `json:"issuer"`
+
+	// Info required to create a credential from a credential application.
+	Credentials []CredentialTemplate `json:"credentials"`
+}
+
+type GetIssuanceTemplateResponse struct {
+	// The template that was requested.
+	IssuanceTemplate IssuanceTemplate `json:"issuanceTemplate"`
+}
+
+type CreateIssuanceTemplateRequest struct {
```

Comment:
In most other places we don't let the user set the ID, we create a new ID with the object newly created object.

Is the flow for this needed where the user needs to create a unique id?

Do we have plans on how to handle it if the id is already taken

Reply:
Excellent point!

I made some changes so that the ID is not required for the endpoint (this is the router one), but still is required in the service.